dhall-docs/src/Dhall/Docs/Html.hs

```diff
@@ -121,17 +127,71 @@ indexToHtml indexDir files dirs params@DocParams{..} = doctypehtml_ $ do
 
     listDir :: Path Rel Dir -> Html ()
     listDir dir =
-        let filePath = Data.Text.pack $ Path.fromRelDir dir in
-        li_ $ a_ [href_ (filePath <> "index.html")] $ toHtml filePath
+        let dirPath = Data.Text.pack $ Path.fromRelDir dir in
+        li_ $ a_ [href_ (dirPath <> "index.html")] $ toHtml dirPath
 
     tryToTakeExt :: Path Rel File -> FilePath
 
     tryToTakeExt file = Path.fromRelFile $ case Path.splitExtension file of
         Nothing -> file
         Just (f, _) -> f
 
-    title :: String
-    title = indexDir <> " index"
+    breadcrumbs = relPathToBreadcrumb indexDir
+    htmlTitle = breadCrumbsToText breadcrumbs
+    pageTitle = breadCrumbsToHtml Index breadcrumbs
+
+-- | ADT for handling bread crumbs. This is essentially a backwards list
```

Comment:
Were there any issues with using a non-backwards list?

Reply:
Yes, the main issue is how to calculate the `../` path (which ends up being different depending if the file is an `index.html` or a `.dhall.html` file). If it were a non-backwards list, then:

* it might need an extra attribute, reflecting how many times do I have to `replicate` `../`, or;
* Traverse the whole breadcrumbs to calculate this in order to add another function helper to convert it to html with links set-up properly.

I did so and I didn't like how the final code, so I changed to a backwards list:

* I don't need to add any other attribute. The breadcrumbs is just a semantic list-like ADT.
* I can start at either 0 or -1 depending if breadcrumb is for an index.html or `.dhall.html` file, and keep incrementing that number at the same time we traverse it.

Reply:
Alright, that sounds reasonable to me